### DISCUSSION TITLE
How can the state be changed in a consumers-side component using `@lit/context`?

### DISCUSSION THREAD

**Discussion by ouweiya** [lit/lit#4507] [Q&A]
I'm trying to change the state in the consumer component, but no rendering occurs. What is the correct way to update values in `@lit/context`?

[Lit Playground Demo Link](https://lit.dev/playground/#sample=examples%2Fcontext-basics&project=W3sibmFtZSI6ImVsZW1lbnRzLnRzIiwiY29udGVudCI6ImltcG9ydCB7aHRtbCwgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7Y3VzdG9tRWxlbWVudCwgcHJvcGVydHl9IGZyb20gJ2xpdC9kZWNvcmF0b3JzLmpzJztcbmltcG9ydCB7cHJvdmlkZSwgY29uc3VtZSwgY3JlYXRlQ29udGV4dH0gZnJvbSAnQGxpdC9jb250ZXh0JztcblxuZXhwb3J0IGNvbnN0IG15Q29udGV4dCA9IGNyZWF0ZUNvbnRleHQ8bnVtYmVyPihTeW1ib2woJ215LWNvbnRleHQnKSk7XG5cbkBjdXN0b21FbGVtZW50KCdteS1wYXJlbicpXG5jbGFzcyBNeVBhcmVudCBleHRlbmRzIExpdEVsZW1lbnQge1xuICAgIEBwcm92aWRlKHsgY29udGV4dDogbXlDb250ZXh0IH0pXG4gICAgQHByb3BlcnR5KHsgdHlwZTogTnVtYmVyIH0pXG4gICAgY291bnQgPSAxO1xuXG4gICAgcmVuZGVyKCkge1xuICAgICAgICByZXR1cm4gaHRtbGBcbiAgICAgICAgICAgIDxkaXY-XG4gICAgICAgICAgICAgICAgPGgxPiR7dGhpcy5jb3VudH08L2gxPlxuICAgICAgICAgICAgICAgIDxteS1jaGlsZD48L215LWNoaWxkPlxuICAgICAgICAgICAgPC9kaXY-XG4gICAgICAgIGA7XG4gICAgfVxufVxuXG5AY3VzdG9tRWxlbWVudCgnbXktY2hpbGQnKVxuY2xhc3MgTXlDaGlsZCBleHRlbmRzIExpdEVsZW1lbnQge1xuICAgIEBjb25zdW1lKHsgY29udGV4dDogbXlDb250ZXh0LCBzdWJzY3JpYmU6IHRydWUgfSlcbiAgICBjb3VudDogbnVtYmVyO1xuXG4gICAgaW5jcmVtZW50SGFuZGxlID0gKCkgPT4ge1xuICAgICAgICB0aGlzLmNvdW50Kys7XG4gICAgICAgIGNvbnNvbGUubG9nKHRoaXMuY291bnQpO1xuICAgIH07XG5cbiAgICByZW5kZXIoKSB7XG4gICAgICAgIHJldHVybiBodG1sYFxuICAgICAgICAgICAgPGRpdiBjbGFzcz1cIlwiPlxuICAgICAgICAgICAgICAgIDxoMz4ke3RoaXMuY291bnQgKiAyfTwvaDM-XG4gICAgICAgICAgICAgICAgPGJ1dHRvbiBAY2xpY2s9JHt0aGlzLmluY3JlbWVudEhhbmRsZX0-SW5jcmVtZW50PC9idXR0b24-XG4gICAgICAgICAgICA8L2Rpdj5cbiAgICAgICAgYDtcbiAgICB9XG59XG4ifSx7Im5hbWUiOiJpbmRleC5odG1sIiwiY29udGVudCI6IjwhRE9DVFlQRSBodG1sPlxuPHNjcmlwdCB0eXBlPVwibW9kdWxlXCIgc3JjPVwiLi9lbGVtZW50cy5qc1wiPjwvc2NyaXB0PlxuIFxuPG15LXBhcmVuPjwvbXktcGFyZW4-XG4gIn0seyJuYW1lIjoicGFja2FnZS5qc29uIiwiY29udGVudCI6IntcbiAgXCJkZXBlbmRlbmNpZXNcIjoge1xuICAgIFwibGl0XCI6IFwiXjMuMC4wXCIsXG4gICAgXCJAbGl0L3JlYWN0aXZlLWVsZW1lbnRcIjogXCJeMi4wLjBcIixcbiAgICBcImxpdC1lbGVtZW50XCI6IFwiXjQuMC4wXCIsXG4gICAgXCJsaXQtaHRtbFwiOiBcIl4zLjAuMFwiXG4gIH1cbn0iLCJoaWRkZW4iOnRydWV9XQ)

```ts
import {html, LitElement} from 'lit';
import {customElement, property} from 'lit/decorators.js';
import {provide, consume, createContext} from '@lit/context';

export const myContext = createContext<number>(Symbol('my-context'));

@customElement('my-paren')
class MyParent extends LitElement {
    @provide({ context: myContext })
    @property({ type: Number })
    count = 1;

    render() {
        return html`
            <div>
                <h1>${this.count}</h1>
                <my-child></my-child>
            </div>
        `;
    }
}

@customElement('my-child')
class MyChild extends LitElement {
    @consume({ context: myContext, subscribe: true })
    count: number;

    incrementHandle = () => {
        this.count++;
        console.log(this.count);
    };

    render() {
        return html`
            <div class="">
                <h3>${this.count * 2}</h3>
                <button @click=${this.incrementHandle}>Increment</button>
            </div>
        `;
    }
}
```

I tried sharing the event function as a state value to other components, which renders successfully, but what is the best way to change the state of `@lit/context`?

[Modified Demo Playground](https://lit.dev/playground/#sample=examples%2Fcontext-basics&project=W3sibmFtZSI6ImVsZW1lbnRzLnRzIiwiY29udGVudCI6ImltcG9ydCB7aHRtbCwgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7Y3VzdG9tRWxlbWVudCwgcHJvcGVydHl9IGZyb20gJ2xpdC9kZWNvcmF0b3JzLmpzJztcbmltcG9ydCB7cHJvdmlkZSwgY29uc3VtZSwgY3JlYXRlQ29udGV4dH0gZnJvbSAnQGxpdC9jb250ZXh0JztcblxuZXhwb3J0IGNvbnN0IG15Q29udGV4dCA9IGNyZWF0ZUNvbnRleHQ8bnVtYmVyPihTeW1ib2woJ215LWNvbnRleHQnKSk7XG5leHBvcnQgY29uc3QgbXlDb250ZXh0SGFuZGxlID0gY3JlYXRlQ29udGV4dDwoczogc3RyaW5nKSA9PiB2b2lkPihTeW1ib2woJ215LWNvbnRleHQtaGFuZGxlJykpO1xuXG5AY3VzdG9tRWxlbWVudCgnbXktcGFyZW4nKVxuY2xhc3MgTXlQYXJlbnQgZXh0ZW5kcyBMaXRFbGVtZW50IHtcbiAgICBAcHJvdmlkZSh7IGNvbnRleHQ6IG15Q29udGV4dCB9KVxuICAgIEBwcm9wZXJ0eSh7IHR5cGU6IE51bWJlciB9KVxuICAgIGNvdW50ID0gMTtcblxuICAgIEBwcm92aWRlKHsgY29udGV4dDogbXlDb250ZXh0SGFuZGxlIH0pXG4gICAgaW5jcmVtZW50SGFuZGxlID0gKCkgPT4ge1xuICAgICAgICB0aGlzLmNvdW50Kys7XG4gICAgICAgIGNvbnNvbGUubG9nKHRoaXMuY291bnQpO1xuICAgIH07XG5cbiAgICByZW5kZXIoKSB7XG4gICAgICAgIHJldHVybiBodG1sYFxuICAgICAgICAgICAgPGRpdj5cbiAgICAgICAgICAgICAgICA8aDE-JHt0aGlzLmNvdW50fTwvaDE-XG4gICAgICAgICAgICAgICAgPG15LWNoaWxkPjwvbXktY2hpbGQ-XG4gICAgICAgICAgICA8L2Rpdj5cbiAgICAgICAgYDtcbiAgICB9XG59XG5cbkBjdXN0b21FbGVtZW50KCdteS1jaGlsZCcpXG5jbGFzcyBNeUNoaWxkIGV4dGVuZHMgTGl0RWxlbWVudCB7XG4gICAgQGNvbnN1bWUoeyBjb250ZXh0OiBteUNvbnRleHQsIHN1YnNjcmliZTogdHJ1ZSB9KVxuICAgIGNvdW50OiBudW1iZXI7XG5cbiAgICBAY29uc3VtZSh7IGNvbnRleHQ6IG15Q29udGV4dEhhbmRsZSwgc3Vic2NyaWJlOiB0cnVlIH0pXG4gICAgaW5jcmVtZW50SGFuZGxlOiAoczogc3RyaW5nKSA9PiB2b2lkO1xuXG4gICAgcmVuZGVyKCkge1xuICAgICAgICByZXR1cm4gaHRtbGBcbiAgICAgICAgICAgIDxkaXYgY2xhc3M9XCJcIj5cbiAgICAgICAgICAgICAgICA8aDM-JHt0aGlzLmNvdW50ICogMn08L2gzPlxuICAgICAgICAgICAgICAgIDxidXR0b24gQGNsaWNrPSR7dGhpcy5pbmNyZW1lbnRIYW5kbGV9PkluY3JlbWVudDwvYnV0dG9uPlxuICAgICAgICAgICAgPC9kaXY-XG4gICAgICAgIGA7XG4gICAgfVxufVxuXG4ifSx7Im5hbWUiOiJpbmRleC5odG1sIiwiY29udGVudCI6IjwhRE9DVFlQRSBodG1sPlxuPHNjcmlwdCB0eXBlPVwibW9kdWxlXCIgc3JjPVwiLi9lbGVtZW50cy5qc1wiPjwvc2NyaXB0PlxuIFxuPG15LXBhcmVuPjwvbXktcGFyZW4-XG4gIn0seyJuYW1lIjoicGFja2FnZS5qc29uIiwiY29udGVudCI6IntcbiAgXCJkZXBlbmRlbmNpZXNcIjoge1xuICAgIFwibGl0XCI6IFwiXjMuMC4wXCIsXG4gICAgXCJAbGl0L3JlYWN0aXZlLWVsZW1lbnRcIjogXCJeMi4wLjBcIixcbiAgICBcImxpdC1lbGVtZW50XCI6IFwiXjQuMC4wXCIsXG4gICAgXCJsaXQtaHRtbFwiOiBcIl4zLjAuMFwiXG4gIH1cbn0iLCJoaWRkZW4iOnRydWV9XQ)

```ts
export const myContext = createContext<number>(Symbol('my-context'));
export const myContextHandle = createContext<(s: string) => void>(Symbol('my-context-handle'));

@customElement('my-paren')
class MyParent extends LitElement {
    @provide({ context: myContext })
    @property({ type: Number })
    count = 1;

    @provide({ context: myContextHandle })
    incrementHandle = () => {
        this.count++;
        console.log(this.count);
    };

    render() {
        return html`
            <div>
                <h1>${this.count}</h1>
                <my-child></my-child>
            </div>
        `;
    }
}

@customElement('my-child')
class MyChild extends LitElement {
    @consume({ context: myContext, subscribe: true })
    count: number;

    @consume({ context: myContextHandle, subscribe: true })
    incrementHandle: (s: string) => void;

    render() {
        return html`
            <div class="">
                <h3>${this.count * 2}</h3>
                <button @click=${this.incrementHandle}>Increment</button>
            </div>
        `;
    }
}
```

I've tried changing the state through custom events, which can successfully update the value, but what approach is in line with the design of `@lit/context`?

[Custom Events Playground](https://lit.dev/playground/#sample=examples%2Fcontext-basics&project=W3sibmFtZSI6ImVsZW1lbnRzLnRzIiwiY29udGVudCI6ImltcG9ydCB7aHRtbCwgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7Y3VzdG9tRWxlbWVudCwgcHJvcGVydHl9IGZyb20gJ2xpdC9kZWNvcmF0b3JzLmpzJztcbmltcG9ydCB7cHJvdmlkZSwgY29uc3VtZSwgY3JlYXRlQ29udGV4dH0gZnJvbSAnQGxpdC9jb250ZXh0JztcblxuZXhwb3J0IGNvbnN0IG15Q29udGV4dCA9IGNyZWF0ZUNvbnRleHQ8bnVtYmVyPihTeW1ib2woJ215LWNvbnRleHQnKSk7XG5cbkBjdXN0b21FbGVtZW50KCdteS1wYXJlbicpXG5jbGFzcyBNeVBhcmVudCBleHRlbmRzIExpdEVsZW1lbnQge1xuICAgIEBwcm92aWRlKHsgY29udGV4dDogbXlDb250ZXh0IH0pXG4gICAgQHByb3BlcnR5KHsgdHlwZTogTnVtYmVyIH0pXG4gICAgY291bnQgPSAxO1xuXG4gICAgY29ubmVjdGVkQ2FsbGJhY2soKSB7XG4gICAgICAgIHN1cGVyLmNvbm5lY3RlZENhbGxiYWNrKCk7XG4gICAgICAgIHRoaXMuYWRkRXZlbnRMaXN0ZW5lcignbXktZXZlbnQnLCB0aGlzLmluY3JlbWVudEhhbmRsZSk7XG4gICAgfVxuICBcbiAgICBpbmNyZW1lbnRIYW5kbGUgPSAoKSA9PiB7XG4gICAgICAgIHRoaXMuY291bnQrKztcbiAgICAgICAgY29uc29sZS5sb2codGhpcy5jb3VudCk7XG4gICAgfTtcblxuICAgIHJlbmRlcigpIHtcbiAgICAgICAgcmV0dXJuIGh0bWxgXG4gICAgICAgICAgICA8ZGl2PlxuICAgICAgICAgICAgICAgIDxoMT4ke3RoaXMuY291bnR9PC9oMT5cbiAgICAgICAgICAgICAgICA8bXktY2hpbGQ-PC9teS1jaGlsZD5cbiAgICAgICAgICAgIDwvZGl2PlxuICAgICAgICBgO1xuICAgIH1cbn1cblxuQGN1c3RvbUVsZW1lbnQoJ215LWNoaWxkJylcbmNsYXNzIE15Q2hpbGQgZXh0ZW5kcyBMaXRFbGVtZW50IHtcbiAgICBAY29uc3VtZSh7IGNvbnRleHQ6IG15Q29udGV4dCwgc3Vic2NyaWJlOiB0cnVlIH0pXG4gICAgY291bnQ6IG51bWJlcjtcblxuICAgIHByaXZhdGUgaW5jcmVtZW50SGFuZGxlID0gKGU6IEV2ZW50KSA9PiB7XG4gICAgICAgIGNvbnN0IGV2ZW50ID0gbmV3IEN1c3RvbUV2ZW50PHVuZGVmaW5lZD4oJ215LWV2ZW50JywgeyBidWJibGVzOiB0cnVlLCBjb21wb3NlZDogdHJ1ZSB9KTtcbiAgICAgICAgdGhpcy5kaXNwYXRjaEV2ZW50KGV2ZW50KTtcbiAgICB9O1xuICBcbiAgICByZW5kZXIoKSB7XG4gICAgICAgIHJldHVybiBodG1sYFxuICAgICAgICAgICAgPGRpdiBjbGFzcz1cIlwiPlxuICAgICAgICAgICAgICAgIDxoMz4ke3RoaXMuY291bnQgKiAyfTwvaDM-XG4gICAgICAgICAgICAgICAgPGJ1dHRvbiBAY2xpY2s9JHt0aGlzLmluY3JlbWVudEhhbmRsZX0-SW5jcmVtZW50PC9idXR0b24-XG4gICAgICAgICAgICA8L2Rpdj5cbiAgICAgICAgYDtcbiAgICB9XG59XG5cbiJ9LHsibmFtZSI6ImluZGV4Lmh0bWwiLCJjb250ZW50IjoiPCFET0NUWVBFIGh0bWw-XG48c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL2VsZW1lbnRzLmpzXCI-PC9zY3JpcHQ-XG4gXG48bXktcGFyZW4-PC9teS1wYXJlbj5cbiAifSx7Im5hbWUiOiJwYWNrYWdlLmpzb24iLCJjb250ZW50Ijoie1xuICBcImRlcGVuZGVuY2llc1wiOiB7XG4gICAgXCJsaXRcIjogXCJeMy4wLjBcIixcbiAgICBcIkBsaXQvcmVhY3RpdmUtZWxlbWVudFwiOiBcIl4yLjAuMFwiLFxuICAgIFwibGl0LWVsZW1lbnRcIjogXCJeNC4wLjBcIixcbiAgICBcImxpdC1odG1sXCI6IFwiXjMuMC4wXCJcbiAgfVxufSIsImhpZGRlbiI6dHJ1ZX1d)

```ts
export const myContext = createContext<number>(Symbol('my-context'));

@customElement('my-paren')
class MyParent extends LitElement {
    @provide({ context: myContext })
    @property({ type: Number })
    count = 1;

    connectedCallback() {
        super.connectedCallback();
        this.addEventListener('my-event', this.incrementHandle);
    }
  
    incrementHandle = () => {
        this.count++;
        console.log(this.count);
    };

    render() {
        return html`
            <div>
                <h1>${this.count}</h1>
                <my-child></my-child>
            </div>
        `;
    }
}

@customElement('my-child')
class MyChild extends LitElement {
    @consume({ context: myContext, subscribe: true })
    count: number;

    private incrementHandle = (e: Event) => {
        const event = new CustomEvent<undefined>('my-event', { bubbles: true, composed: true });
        this.dispatchEvent(event);
    };
  
    render() {
        return html`
            <div class="">
                <h3>${this.count * 2}</h3>
                <button @click=${this.incrementHandle}>Increment</button>
            </div>
        `;
    }
}
```

**Comment by augustjk** [comment]
Both are valid approaches.

It is possible to combine the value and the updater into a single context like this: https://lit.dev/playground/#sample=examples%2Fcontext-basics&gist=5aa7d41fe7a1f3132601a9363e70879d

I'd prefer this over having to manage a separate context just for the updater.

But using events is also quite idiomatic for web components so using it to update the value seems fine. Depending on how other providers are configured, you may want to stop propagation of the event so it doesn't update any providers further up.

**Reply by herrKlein** [reply]
I'm also trying different approaches. I like this approach except for the fact that I can't use my-child as a single instance ( without its parent )
example here: [https://lit.dev/playground/#sample=examples%2Fcontext-basics&gist=bf79214b366a9476bf6f38743c363754](https://lit.dev/playground/#sample=examples%2Fcontext-basics&gist=bf79214b366a9476bf6f38743c363754)

How would you go, having the child working as a single instance, and updating also its parent if its available. The use case for us is, we dynamically add different child components, they should register and the parent should also have their state. So when removing and adding the child components the state can be retained.

We also want those child components function on their own.

**Reply by augustjk** [reply]
There's probably a better way to organize this but a quick stab would be something like this: https://lit.dev/playground/#sample=examples%2Fcontext-basics&gist=ff90d62806f2f1752d818ab86ce93086

Using a custom accessor to either use the context, if present, or default to an internal property.

Edit to add: Other things can be done depending on requirements, like should the child component be able to also take the count value as a property.

**Reply by herrKlein** [reply]
Thank, I will try this and maybe make a more comprehensive example for what I need. adding and removing child components, while their state is retained by the provider. updated by the child and given to the child when its added again.

**Comment by ouweiya** [comment]
> Both are valid approaches.
> 
> It is possible to combine the value and the updater into a single context like this: https://lit.dev/playground/#sample=examples%2Fcontext-basics&gist=5aa7d41fe7a1f3132601a9363e70879d
> 
> I'd prefer this over having to manage a separate context just for the updater.
> 
> But using events is also quite idiomatic for web components so using it to update the value seems fine. Depending on how other providers are configured, you may want to stop propagation of the event so it doesn't update any providers further up.


Unfortunately, it seems that the provided playground link is incorrect. Please upload it again. Thank you.

![image](https://github.com/lit/lit/assets/28357731/3c283500-e738-4949-bd21-c709898349e6)

**Reply by augustjk** [reply]
That's odd. It works fine on my end. Maybe it's some GitHub issue.
Here's a [long link](https://lit.dev/playground/#sample=examples%2Fcontext-basics&project=W3sibmFtZSI6ImVsZW1lbnRzLnRzIiwiY29udGVudCI6ImltcG9ydCB7aHRtbCwgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7Y3VzdG9tRWxlbWVudCwgcHJvcGVydHl9IGZyb20gJ2xpdC9kZWNvcmF0b3JzLmpzJztcbmltcG9ydCB7cHJvdmlkZSwgY29uc3VtZSwgY3JlYXRlQ29udGV4dH0gZnJvbSAnQGxpdC9jb250ZXh0JztcblxuaW50ZXJmYWNlIENvdW50Q29udGV4dCB7XG4gIHZhbHVlOiBudW1iZXI7XG4gIGluY3JlbWVudDogKCkgPT4gdm9pZDtcbn1cblxuZXhwb3J0IGNvbnN0IG15Q29udGV4dCA9IGNyZWF0ZUNvbnRleHQ8Q291bnRDb250ZXh0PihTeW1ib2woJ215LWNvbnRleHQnKSk7XG5cbkBjdXN0b21FbGVtZW50KCdteS1wYXJlbicpXG5jbGFzcyBNeVBhcmVudCBleHRlbmRzIExpdEVsZW1lbnQge1xuICAgIGluY3JlbWVudEhhbmRsZSA9ICgpID0-IHtcbiAgICAgICAgdGhpcy5jb3VudCA9IHtcbiAgICAgICAgICB2YWx1ZTogdGhpcy5jb3VudC52YWx1ZSArIDEsXG4gICAgICAgICAgaW5jcmVtZW50OiB0aGlzLmluY3JlbWVudEhhbmRsZSxcbiAgICAgICAgfTtcbiAgICAgICAgY29uc29sZS5sb2codGhpcy5jb3VudCk7XG4gICAgfTtcblxuICAgIEBwcm92aWRlKHsgY29udGV4dDogbXlDb250ZXh0IH0pXG4gICAgQHByb3BlcnR5KHsgdHlwZTogT2JqZWN0IH0pXG4gICAgY291bnQgPSB7XG4gICAgICB2YWx1ZTogMSxcbiAgICAgIGluY3JlbWVudDogdGhpcy5pbmNyZW1lbnRIYW5kbGVcbiAgICB9O1xuICBcblxuICAgIHJlbmRlcigpIHtcbiAgICAgICAgcmV0dXJuIGh0bWxgXG4gICAgICAgICAgICA8ZGl2PlxuICAgICAgICAgICAgICAgIDxoMT4ke3RoaXMuY291bnQudmFsdWV9PC9oMT5cbiAgICAgICAgICAgICAgICA8bXktY2hpbGQ-PC9teS1jaGlsZD5cbiAgICAgICAgICAgIDwvZGl2PlxuICAgICAgICBgO1xuICAgIH1cbn1cblxuQGN1c3RvbUVsZW1lbnQoJ215LWNoaWxkJylcbmNsYXNzIE15Q2hpbGQgZXh0ZW5kcyBMaXRFbGVtZW50IHtcbiAgICBAY29uc3VtZSh7IGNvbnRleHQ6IG15Q29udGV4dCwgc3Vic2NyaWJlOiB0cnVlIH0pXG4gICAgY291bnQ6IENvdW50Q29udGV4dDtcblxuICAgIHJlbmRlcigpIHtcbiAgICAgICAgcmV0dXJuIGh0bWxgXG4gICAgICAgICAgICA8ZGl2IGNsYXNzPVwiXCI-XG4gICAgICAgICAgICAgICAgPGgzPiR7dGhpcy5jb3VudC52YWx1ZSAqIDJ9PC9oMz5cbiAgICAgICAgICAgICAgICA8YnV0dG9uIEBjbGljaz0ke3RoaXMuY291bnQuaW5jcmVtZW50fT5JbmNyZW1lbnQ8L2J1dHRvbj5cbiAgICAgICAgICAgIDwvZGl2PlxuICAgICAgICBgO1xuICAgIH1cbn1cblxuIn0seyJuYW1lIjoiaW5kZXguaHRtbCIsImNvbnRlbnQiOiI8IURPQ1RZUEUgaHRtbD5cbjxzY3JpcHQgdHlwZT1cIm1vZHVsZVwiIHNyYz1cIi4vZWxlbWVudHMuanNcIj48L3NjcmlwdD5cbiBcbjxteS1wYXJlbj48L215LXBhcmVuPlxuICJ9LHsibmFtZSI6InBhY2thZ2UuanNvbiIsImNvbnRlbnQiOiJ7XG4gIFwiZGVwZW5kZW5jaWVzXCI6IHtcbiAgICBcImxpdFwiOiBcIl4zLjAuMFwiLFxuICAgIFwiQGxpdC9yZWFjdGl2ZS1lbGVtZW50XCI6IFwiXjIuMC4wXCIsXG4gICAgXCJsaXQtZWxlbWVudFwiOiBcIl40LjAuMFwiLFxuICAgIFwibGl0LWh0bWxcIjogXCJeMy4wLjBcIlxuICB9XG59IiwiaGlkZGVuIjp0cnVlfV0) instead.

**Comment by ouweiya** [comment]
Can I use `@lit/context` in Reactive Controllers?

I need to get shared data in Reactive Controllers, but using `@consume({context: xxx})` results in an error.

**Reply by augustjk** [reply]
You can use the `ContextConsumer` controller from within the controller. See https://lit.dev/docs/composition/controllers/#building-controllers-from-other-controllers
